src/backend/routers/paras.ts

```diff
@@ -0,0 +1,59 @@
+import { z } from "zod";
+import { procedure } from "../trpc";
+
+export const paraProcedures = {
+  getParaById: procedure
+    .input(z.object({ user_id: z.string().uuid() }))
+    .query(async (req) => {
+      console.log("req", req);
+      const { user_id } = req.input;
+
+      const result = await req.ctx.db
+        .selectFrom("user")
+        .where("user_id", "=", user_id)
+        .selectAll()
+        .executeTakeFirst();
+
+      return result;
+    }),
+
+  getAllParas: procedure.query(async (req) => {
+    const result = await req.ctx.db.selectFrom("user").selectAll().execute();
+
+    return result;
+  }),
+
+  createPara: procedure
+    .input(
+      z.object({
+        first_name: z.string(),
+        last_name: z.string(),
+        email: z.string(),
+        role: z.string(),
+      })
+    )
+    .mutation(async (req) => {
+      const { first_name, last_name, email, role } = req.input;
+
+      // todo: add a unique constraint to prevent duplicate paras
+      const result = await req.ctx.db
+        .insertInto("user")
+        .values({ first_name, last_name, email, role })
+        .returningAll()
+        .execute();
+      return result;
+    }),
+
+  deletePara: procedure
```

Comment:
"Delete" should really be archive - i.e. remove the link from the CM->para, but not actually delete the para row in the database. If we actually delete the para, all the data that they might have collected now has a dangling user_id.

I would recommend removing this function or changing it to archive.

In general, I don't think we should ever delete (remove database rows) from the frontend - if we want to delete/cleanup things, that can be handled by a "gardener" job that aims to maintain the referential integrity (this is a common pattern in NoSQL world, in SQL world, maybe we can create those constraints in the data model, so we might not need this job)

Reply:
We could do this with a link/unlink or CM ID property if the para has a CM_id involved. It may also be worthwhile considering that a CM's dashboard will only display THEIR paras and the tasks they have assigned to their Para; A CM should ONLY see the tasks they have assigned to THEIR Paras and not a Para's entire task list, if it possibly includes tasks from other CM's. This will require a lot more restructuring, including of the schema, to implement.